torchao/quantization/quant_primitives.py

```diff
@@ -288,17 +298,26 @@ def choose_qparams_affine(
     min_val = torch.amin(input, dim=reduction_dims, keepdim=False)
     max_val = torch.amax(input, dim=reduction_dims, keepdim=False)
 
-    min_val_neg = torch.min(min_val, torch.zeros_like(min_val))
-    max_val_pos = torch.max(max_val, torch.zeros_like(max_val))
+    if preserve_zero:
+        min_val_neg = torch.min(min_val, torch.zeros_like(min_val))
+        max_val_pos = torch.max(max_val, torch.zeros_like(max_val))
+    else:
+        min_val_neg = min_val
+        max_val_pos = max_val
 
     if mapping_type == MappingType.SYMMETRIC:
         max_val_pos = torch.max(-min_val_neg, max_val_pos)
         scale = max_val_pos / (float(quant_max - quant_min) / 2)
+        assert preserve_zero, "non-representable zero path is not implemented for symmetric quantization"
```

Comment:
Adding a `assertRaisesRegex` based test for this as a follow up would be good. Also you can raise a ValueError. This prevents us from accidentally adding coverage later on even though it might mathematically not make sense.

Reply:
OK sounds good